xarray/core/dataarray.py

```diff
@@ -2289,6 +2292,53 @@ def rank(self, dim, pct=False, keep_attrs=False):
         ds = self._to_temp_dataset().rank(dim, pct=pct, keep_attrs=keep_attrs)
         return self._from_temp_dataset(ds)
 
+    def differentiate(self, coord, edge_order=1):
```

Comment:
Shouldn't this be `dim` instead of `coord`? I'm thinking of multidimensional co-ordinates here.

Reply:
Currently, a 1-dimensional (non-dimensional) coordinate is also allowed. Do you think it is confusing?
For a multidimensional coordinate, ValueError will be raised.

Reply:
OK¸ maybe add a little more to the docstring then to make it clear that we expect 1D co-ordinates.